src/releaseNotesProvider.ts

```diff
@@ -43,16 +43,16 @@ async function showReleaseNotesImpl(
 ): Promise<Either<string, void>> {
   const state = context.globalState;
 
-  const remote = isRemote();
-  if (remote.kind === "left") {
-    return remote;
-  }
-
-  const version = getVersion(calledOn);
+  const version = getVersion();
   if (version.kind === "left") {
     return version;
   }
 
+  const remote = isRemote();
+  if (calledOn === "onExtensionStart" && remote.kind === "left") {
+    return remote;
+  }
```

Comment:
Makes more sense to check version first, only after remote. It yields better message: `Release notes was not shown: do not show release notes for an older version/already seen version`